app/grandchallenge/serving/views.py

```diff
@@ -74,3 +75,30 @@ def serve_submissions(request, *, submission_pk, **_):
         )
 
     raise PermissionDenied
+
+
+def serve_component_interface_value(
+    request, *, component_interface_value_pk, **_
+):
+    try:
+        user, _ = TokenAuthentication().authenticate(request)
+    except (AuthenticationFailed, TypeError):
+        user = request.user
+
+    try:
+        # output should only be connected to a single job; throw error if not?
+        civ = (
+            ComponentInterfaceValue.objects.filter(
+                pk=component_interface_value_pk
+            )
+            .prefetch_related("algorithms_jobs_as_output")
+            .first()
+        )
```

Comment:
Rather than doing a filter and then `first()`, it is better to do a `get(pk=component_interface_value_pk)`, catching the `MultipleObjectsReturned` exception if necessary. You can still add the `prefetch_related` before the `get`.

Reply:
But, I think if you implement my comment on L98 then the prefetch related is unnecessary.